hal/mpu_api.h

```diff
@@ -64,8 +64,8 @@ extern "C" {
  * Initialize the MPU
  *
  * Initialize or re-initialize the memory protection unit.
- * It is implementation defined what region are protected
- * by the MPU after initialization.
+ * After initialization or re-initialization, ROM and RAM protection
+ * are both enabled.
```

Comment:
Since this was undefined behavior it wasn't tested. It may be good to add a testcase to ensure that `mbed_mpu_init` enables memory protection.

Reply:
Done.